### PR TITLE
Custom ptsets

### PR DESCRIPTION
**Pull request by anilyil**
## Purpose

This PR adds capability to set custom pointsets when adding DVGeo to the aero solver. The custom pointsets are provided in a dict where keys are the corresponding family names in the solver and values are the custom kwargs passed with those pointsets. See the relevant ADflow PR: https://github.com/mdolab/adflow/pull/299

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Comment by marcomangano** [comment]
LGTM, I opened a separate PR for the flake8 issues. Once that is merged this is good to go, so we can move on to the related ADflow PR.

File: baseclasses/solvers/pyAero_solver.py [hunk 1]
```diff
@@ -95,6 +96,13 @@ def setDVGeo(self, DVGeo, pointSetKwargs=None):
             Keyword arguments to be passed to the DVGeo addPointSet call.
             Useful for DVGeometryMulti, specifying FFD projection tolerances, etc.
 
+        customPointSetKwargs : dict of dicts
```
[the remark this answers is not in the document]
Comment (A-CGray) [reply baseclasses/solvers/pyAero_solver.py:99]: OK, I'm fine with just the docstring changes

**Review by A-CGray — DISMISSED** [review]

File: baseclasses/solvers/pyAero_solver.py [hunk 2]
```diff
@@ -103,10 +114,11 @@ def setDVGeo(self, DVGeo, pointSetKwargs=None):
 
         self.DVGeo = DVGeo
 
-        if pointSetKwargs is None:
-            self.pointSetKwargs = {}
-        else:
-            self.pointSetKwargs = pointSetKwargs
```
Comment (sseraj) [inline baseclasses/solvers/pyAero_solver.py:109]: This code avoids [mutable defaults](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/), so we should keep it like this.

File: baseclasses/solvers/pyAero_solver.py [hunk 3]
```diff
@@ -94,6 +95,16 @@ def setDVGeo(self, DVGeo, pointSetKwargs=None):
         pointSetKwargs : dict
             Keyword arguments to be passed to the DVGeo addPointSet call.
             Useful for DVGeometryMulti, specifying FFD projection tolerances, etc.
+            These arguments are used for all point sets added by this solver.
+
+        customPointSetFamilies : dict of dicts
+            Keyword arguments to be passed to the DVGeo addPointSet call for each surface family,
+            specified by the keys. The surface families need to be all part of the designSurfaceFamily.
+            Useful for DVGeometryMulti, specifying FFD projection tolerances, etc.
```
Comment (sseraj) [inline baseclasses/solvers/pyAero_solver.py:103]: I feel like the three lines above do not emphasize the main purpose of this argument, which is to have subsets of the CFD surface as separate point sets. I would rephrase this to first talk about what the keys are and what they do, then talk about how the values are dictionaries of keyword arguments.

**Review by sseraj — APPROVED** [review]

**Review by A-CGray — APPROVED** [review]
